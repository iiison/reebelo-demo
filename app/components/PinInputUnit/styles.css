@import "../../styles/base/_variables";

.pinInput {
    outline: none;
    width: 50px;
    height: 70px;
    margin: 5px;
    border: none;
    border: 1px solid;
    border-radius: 10px;
    border-color: grey;
    font-size: x-large;

    .error {
        border-color: $negative;
        color: $negative;
    }

    .success {
        border: $positive;
        color: $positive;
    }
}
