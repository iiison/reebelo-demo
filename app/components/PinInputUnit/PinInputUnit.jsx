import React from 'react'
import { PropTypes } from 'prop-types'

import styles from './styles.css'

const PinInputUnit = ({
  refVal,
  events = {},
  value = '',
  type = 'tel',
  // classes = '',
}) => {
  // TODO: Need to change type from  tel to password on focusout
  return (
    <input
      {...events}
      ref={refVal}
      type={type}
      min={0}
      max={9}
      maxLength={1}
      value={value}
      className={`t-center ${styles.pinInput}`}
    />
  )
}

PinInputUnit.propTypes = {
  events: PropTypes.shape({
    onFocus: PropTypes.func,
    onKeyUp: PropTypes.func,
    onBlur: PropTypes.func,
    onChange: PropTypes.func.isRequired
  }),
  refVal: PropTypes.oneOfType([
    PropTypes.func,
    PropTypes.shape({ current: PropTypes.any })
  ]).isRequired,
  value: PropTypes.string,
  classes: PropTypes.string,
  type: PropTypes.string
}

export default PinInputUnit

