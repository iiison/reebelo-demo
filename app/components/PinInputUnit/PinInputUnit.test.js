import React from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'

import PinInputUnit from './PinInputUnit'


const defaultInputProps = {
  refVal: () => { },
  value: '',
  events: {
    onChange: () => { }
  },
  classes: ''
}

describe('>> COMPONENT -- Tests PinInputUnit', () => {
  it('PinComponent Snapshot Test', () => {
    const props = defaultInputProps
    const component = renderer
      .create(<PinInputUnit {...props} />)
      .toJSON()

    expect(component).toMatchSnapshot();
  })

  it('renders single pin input with default props', () => {
    const props = defaultInputProps
    const input = shallow(<PinInputUnit {...props} />)

    const expectedClass = 'pinInput'

    expect(input.hasClass(expectedClass)).toBe(true)
  })

  // TODO: Is it required ?
  it('renders single pin input with custom class name', () => {
    const props = {
      ...defaultInputProps,
      classes: 'pinInput'
    }
    const input = shallow(<PinInputUnit {...props} />)

    const expectedClass = props.classes
    const expectedType = 'tel'
    const expectedMinVal = 0
    const expectedMaxVal = 9
    const expectedLength = 1

    expect(input.hasClass(expectedClass)).toBe(true)
    expect(input.props().type).toBe(expectedType)
    expect(input.props().min).toBe(expectedMinVal)
    expect(input.props().max).toBe(expectedMaxVal)
    expect(input.props().maxLength).toBe(expectedLength)
  })

  it('renders single pin input with custom value', () => {
    const props = {
      ...defaultInputProps,
      value: 'some value'
    }
    const input = shallow(<PinInputUnit {...props} />)

    const expectedValue = props.value
    const result = input.props().value

    expect(result).toEqual(expectedValue)
  })

  it('triggers single pin input onChange function', () => {
    const onChangeMock = jest.fn()
    const props = {
      ...defaultInputProps,
      events: {
        onChange: onChangeMock
      }
    }
    const input = shallow(<PinInputUnit {...props} />)

    input.simulate('change', { target: { value: 'My new value' } });

    expect(onChangeMock).toBeCalledTimes(1)
  })
})

