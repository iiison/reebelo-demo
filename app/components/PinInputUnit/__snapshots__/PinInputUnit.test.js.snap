// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>> COMPONENT -- Tests PinInputUnit PinComponent Snapshot Test 1`] = `
<input
  className="t-center pinInput"
  max={9}
  maxLength={1}
  min={0}
  onChange={[Function]}
  type="tel"
  value=""
/>
`;
